go.mod

```diff
@@ -4,6 +4,7 @@ go 1.12
 
 require (
 	github.com/burntsushi/toml v0.3.1
+	github.com/cespare/xxhash v1.1.0
```

Comment:
Why would we use a different hashing algorithm than the one we already have? This adds a dependency and increases complexity. 

Also, we have tested the `fnv1a` algorithm but we never tested this one.

Do we have evidence about the advantages this algorithm provides?

Reply:
Reason being that this in this case, we care about speed. And xxHash is fastest one around. 
https://cyan4973.github.io/xxHash/ .Also, refer to https://medium.com/@dgryski/consistent-hashing-algorithmic-tradeoffs-ef6b8e2fcae8.

Reply:
I will check this in a bit more detail.

Reply:
Yes, I think it should be faster in our context too. Let's keep it after we resolve the copying problem.